### DISCUSSION TITLE
Remove field entirely from result list (via hook)

### DISCUSSION THREAD

**Discussion by kendallroth** [pocketbase/pocketbase#5466] [Q&A] [answered]
The `onRecordsListRequest` hook allow modifying the resulting data set _before_ it is returned to the client. However, I have not found a way to entirely remove a field from the data, as setting to both `undefined` and `null` just make it `""` (empty string). I have tried both `record.set(key, value)` as well as `record.load({ ... })`. Any ideas?

The use case would be that I want to add some fields to a record that one user type cannot see, but another user type can (ie. for sorting purposes, etc). While setting the field to `""` is better than including in the response, it would be nice to entirely remove it from the response. For example, removing the field `reviewScore` entirely for users who should not know that datapoint 🤷.

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not entirely sure that I understand your use case but it is not possible at the moment. You could also check https://github.com/pocketbase/pocketbase/discussions/5369.

**Reply by kendallroth** [reply]
Thanks, I wish I had come across that discussion earlier 🤦‍♂️. That is the same use case effectively; thanks for pointing me to it.
